### PR TITLE
Nickname Topic

### PR DESCRIPTION
**Pull request by Olliesama**
Topic that lets you change Monika's name.

### CODE REVIEW

File: Monika After Story/game/definitions.rpy [hunk 1]
```diff
@@ -2546,6 +2546,7 @@ default persistent.clearall = None
 default persistent.menu_bg_m = None
 default persistent.first_load = None
 default persistent.has_merged = False
+default persistent._mas_affection_nickname = "Monika"
```
Comment (ThePotatoGuy) [inline Monika After Story/game/definitions.rpy]: `_mas_monika_nickname` would be more appropriate as a name.

File: Monika After Story/game/script-affection.rpy [hunk 2]
```diff
@@ -166,6 +166,104 @@ init python:
...
+            "Scum", 
+            "Murder",
+            "Bitch",
+            "Fuck.",
+            "Bully",
+            "Bulli", 
+            "Yuri",
+            "Sayori",
+            "Natsuki"
+            ]
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-affection.rpy]: Fix indentation please

File: Monika After Story/game/script-affection.rpy [hunk 3]
```diff
@@ -166,6 +166,104 @@ init python:
...
+
+        good_nickname_list = [
+            "love",
+            "sweet",
+            "cute",
+            "heart",
+            "darling",
+            "beautiful",
+            "honey"
+                ]
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-affection.rpy]: here too

File: Monika After Story/game/script-affection.rpy [hunk 4]
```diff
@@ -166,6 +166,104 @@ init python:
         #elif datetime.datetime.now() > persistent.sessions["last_session_end"] + datetime.timedelta(days = 4) and datetime.datetime.now() < persistent.sessions["last_session_end"] + datetime.timedelta(days = 14):
 
 
+$ import re
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-affection.rpy]: Move the import inside of this label.

File: Monika After Story/game/script-affection.rpy [hunk 5]
```diff
@@ -166,6 +166,104 @@ init python:
...
+                elif inputname == player:
+                    m 1c "..."
+                    m 1m "That's your name [player]! Give me my own!"
+                    m 1a "Try again~"
+                elif inputname == m_name:
+                    m 1c "..."
+                    m 1l "I thought we were choosing a new name, silly."
+                    m 1e "Try again~"
+                else:
+                    $ nickname = re.compile('|'.join(bad_nickname_list), re.IGNORECASE)
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-affection.rpy]: compiles are costly. You should compile these regexes before the loop.

**Comment by Olliesama** [comment]
Hopefully that's better for you

File: Monika After Story/game/script-affection.rpy [hunk 6]
```diff
@@ -167,6 +167,88 @@ init python:
...
+                        else:
+                            m 3j "From now on, I'll be called {i}'[m_name]'{/i}, ehehe~"
+                        $ done = True
+                    else:
+                        #$ mas_loseAffection(persistent._mas_affection["badexp"])
+                        m 1h "[player]! That's not nice at all!" 
+                        m 2i "This was supposed to be something nice for both of us...why would you say such things?" 
+                        m 1o "If you don't like the idea you should have just said so."
+                        m 3q "...I don't like this idea any more..."
+                        $ hideEventLabel("monika_affection_nickname",lock=False,depool=False)
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-affection.rpy]: aha good. This is quality realism.

**Pull request merged**